src/NosCore.GameObject/Networking/LoginService/LoginService.cs

```diff
@@ -87,12 +88,12 @@ public class LoginService : ILoginService
                     return;
                 }
 
-                if(useApiAuth)
+                if (useApiAuth)
                 {
                     username = await _authHttpClient.GetAwaitingConnectionAsync(null, passwordToken, clientSession.SessionId).ConfigureAwait(false);
                 }
 
-                var acc = _accountDao.FirstOrDefault(s => s.Name.ToLower() == username!.ToLower());
+                var acc = await _accountDao.FirstOrDefaultAsync(s => s.Name.ToLower() == (username ?? "").ToLower()).ConfigureAwait(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5326532)